libqtile/widget/check_updates.py

```diff
@@ -56,29 +56,44 @@ def __init__(self, **config):
                          "Mandriva": ("urpmq --auto-select", 0)
                          }
 
-        # Check if distro name is valid.
-        try:
-            self.cmd = self.cmd_dict[self.distro][0].split()
-            self.subtr = self.cmd_dict[self.distro][1]
-        except KeyError:
-            distros = sorted(self.cmd_dict.keys())
-            logger.error(self.distro + ' is not a valid distro name. ' +
-                         'Use one of the list: ' + str(distros) + '.')
-            self.cmd = None
+        # Backwards compatibility for custom_command. Convert string to tuple.
+        if type(self.custom_command) is not tuple:
+            self.custom_command = (self.custom_command, 0)
+            logger.warning("string inputs to custom_command are deprecated." +
+                           "please use tuple: (\"command\",int(lines to strip from output))")
+
+        if not self.custom_command[0]:
+            # Check if distro name is valid.
+            try:
+                # logger.warning("checking updates with distro")
+                self.cmd = self.cmd_dict[self.distro][0]  # .split()
+                self.subtr = self.cmd_dict[self.distro][1]
+                # logger.warning(self.cmd)
+            except KeyError:
+                distros = sorted(self.cmd_dict.keys())
+                logger.error(self.distro + ' is not a valid distro name. ' +
+                             'Use one of the list: ' + str(distros) + '.')
+                self.cmd = None
+                self.subtr = 0
+        else:
+            # Otherwise use custom_command
+            # logger.warning("checking updates with custom_command")
+            self.cmd = self.custom_command[0]  # .split()
+            self.subtr = self.custom_command[1]
+            # logger.warning(self.cmd)
 
         if self.execute:
             self.add_callbacks({'Button1': self.do_execute})
 
     def _check_updates(self):
         # type: () -> str
         try:
-            if self.custom_command is None:
-                updates = self.call_process(self.cmd)
-            else:
-                updates = self.call_process(self.custom_command, shell=True)
-                self.subtr = 0
-        except CalledProcessError:
-            updates = ""
+            # logger.warning("running check update cmd")
+            updates = self.call_process(self.cmd, shell=True)
+            # logger.warning(str(updates))
+        except CalledProcessError as e:
+            # logger.warning(e.output)
+            updates = e.output.decode()
```

Comment:
I do have to pick up the discussion from #2311 here:
`pamac checkupdates` _always_ has a return code of 100. This is annoying because for all other _check for updates_ commands a non-zero exit code indicates an error. This way we might actually count the line length of error messages (don't know if one of the programs emits an error message on `stdout` though). The only alternative I could think of would be to add yet another argument, the expected return code, defaulting to 0. Then we could just do
```py
res = subprocess.run(self.cmd, shell=True, capture_output=True)
if res.returncode == self.expected_return_code:
    updates = result.stdout.decode()
else:
    logger.error('checking for updates with %r failed:\n%s\n%s', self.cmd, res.stderr.decode(), res.stdout.decode())
    updates = ''
```
What do you guys think?

Reply:
@tych0 what do you think about the retcode?

Reply:
I just realized that the pamac specific issue can be bypassed with a simple shell script.
```
#!/bin/bash
OUT=$(pamac checkupdates -q)
echo $out
exit 0
```
This should bypass pamac's error code and still get the output. I don't have any updates right now but a test script that just echos a couple of lines and then exits 1 worked but the output was all on one line. With some tweaking it could work better.

With that in mind I don't think its worth adding exit code handling for what seems to be one outlier (thoughts?). Unless there are multiple other package managers that also output non 0 codes. I think this pr should focus on just giving custom_command feature parity with cmd_dict. At the very least even if we don't go through with it, it's still worth moving the custom_command check to \_init\_

Reply:
> With that in mind I don't think its worth adding exit code handling for what seems to be one outlier 

I agree, and I think that suggesting small custom scripts like yours above is a valid solution to users whose distro's update checker needs a little wrapping.